### PR TITLE
miniupnpc: By default skip invalid, disconnected and non-IGD devices

### PR DESCRIPTION
**Pull request by pali**
Allow to use such IGD devices by new -i option.

### CODE REVIEW

File: miniupnpc/upnpc.c [hunk 1]
```diff
@@ -705,15 +709,19 @@ int main(int argc, char ** argv)
...
 			case 3:
 				printf("UPnP device found. Is it an IGD ? : %s\n", urls.controlURL);
+				if (!ignore) goto invalid;
 				printf("Trying to continue anyway\n");
 				break;
 			default:
 				printf("Found device (igd ?) : %s\n", urls.controlURL);
+				if (!ignore) goto invalid;
 				printf("Trying to continue anyway\n");
+				break;
```
Comment (miniupnp) [inline miniupnpc/upnpc.c:724]: ?

File: miniupnpc/upnpc.c [hunk 2]
```diff
@@ -705,15 +709,19 @@ int main(int argc, char ** argv)
 				break;
 			case 2:
 				printf("Found a (not connected?) IGD : %s\n", urls.controlURL);
+				if (!ignore) goto invalid;
```
Comment (miniupnp) [inline miniupnpc/upnpc.c:712]: I don't like the goto very much ;)

Comment (pali) [reply miniupnpc/upnpc.c:712]: Ok, do you have any suggestion or improvement, how to achieve this change without goto?

File: miniupnpc/upnpc.c (hunk 1 above)
Comment (pali) [reply miniupnpc/upnpc.c:724]: Well, I added break for the last case of switch. But it is not needed.

File: miniupnpc/upnpc.c (hunk 2 above)
Comment (miniupnp) [reply miniupnpc/upnpc.c:712]: it also introduces a memory leak

Comment (pali) [reply miniupnpc/upnpc.c:712]: Now I see it... I will think a bit more about this code.

Comment (miniupnp) [reply miniupnpc/upnpc.c:712]: see 1a1dc0f

**Comment by miniupnp** [comment]
845dc469d6eaf73f3d0e497c21cebe2576d9ac3d

**Comment by pali** [comment]
Thank you! I updated PR

**Pull request merged**